src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -149,7 +149,13 @@ public async Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext
                 throw new InvalidOperationException(string.Format("Cannot create Collection Information for {0} in database {1} with lease {2} in database {3} : {4}", attribute.CollectionName, attribute.DatabaseName, attribute.LeaseCollectionName, attribute.LeaseDatabaseName, ex.Message), ex);
             }
 
-            return new CosmosDBTriggerBinding(parameter, documentCollectionLocation, leaseCollectionLocation, leaseHostOptions, changeFeedOptions, _logger);
+            // -1 is allowed and indicates "infinite retries", but other negative values are not valid
+            if (attribute.RetryCount < -1)
+            {
+                throw new InvalidOperationException($"{nameof(attribute.RetryCount)} value {attribute.RetryCount} is not valid. Value must be zero, positive, or -1 for infinite retries.");
+            }
+
+            return new CosmosDBTriggerBinding(parameter, documentCollectionLocation, leaseCollectionLocation, leaseHostOptions, changeFeedOptions, _logger, attribute.RetryCount);
```

Comment:
We should make `RetryCount` go through the `NameResolver` so it can be adjusted with app settings. I can update that.

Reply:
Thx, I was not familiar with that or may have missed something.